asab/web/authn/oauth.py

```diff
@@ -4,25 +4,24 @@
 import asab.web.rest
 
 
-async def oauthclient_middleware_factory(request, handler, url):
-	authorization = request.headers.get("Authorization", None)
-	if authorization is None:
-		return asab.web.rest.json_response(request, {
-			"result": "AUTHORIZATION-NOT-PROVIDED"
-		})
-
-	async with aiohttp.ClientSession() as session:
-		async with session.get(url, headers={"Authorization": authorization}) as resp:
-			if resp.status == 200:
-				identity = await resp.json()
-				if identity is not None:
-					request.Identity = identity
-					return await handler(request)
-			else:
-				return asab.web.rest.json_response(request, {
-					"result": "AUTHORIZATION-FAILED"
-				})
-
-	return asab.web.rest.json_response(request, {
-		"result": "AUTHORIZATION-ERROR"
-	})
\ No newline at end of file
+def oauthclient_middleware_factory(app, *args, url, **kwargs):
+
+	@aiohttp.web.middleware
+	async def oauthclient_middleware(request, handler):
+
+		authorization = request.headers.get(aiohttp.hdrs.AUTHORIZATION, None)
+		if authorization is None:
+			return await handler(request)
+
+		async with aiohttp.ClientSession() as session:
+			async with session.get(url, headers={"Authorization": authorization}) as resp:
+				if resp.status == 200:
```

Comment:
We will not need to perform this call for every single inbound request, right?

Reply:
We will create a cache for it, the task is in the backlog.

Reply:
👍